go-controller/pkg/ovn/ovn.go

```diff
@@ -555,6 +561,56 @@ func (oc *Controller) WatchNetworkPolicy() error {
 	return err
 }
 
+// WatchEgressFirewall starts the watching of egressfirewall resource and calls
+// back the appropriate handler logic
+func (oc *Controller) WatchEgressFirewall() error {
+	_, err := oc.watchFactory.AddEgressFirewallHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			egressFirewall := obj.(*egressfirewall.EgressFirewall)
+			errList := oc.addEgressFirewall(egressFirewall)
+			for _, err := range errList {
+				klog.Error(err)
+				egressFirewall.Status.EgressFirewallObjectStatus = "EgressFirewall not applied correctly, please check logs"
+			}
+			if errList == nil {
+				egressFirewall.Status.EgressFirewallObjectStatus = "EgressFirewall Rules applied"
+			}
+			err := oc.updateEgressFirewallObjectWithRetry(egressFirewall)
+			if err != nil {
+				klog.Error(err)
+			}
+		},
+		UpdateFunc: func(old, newer interface{}) {
+			newEgressFirewall := newer.(*egressfirewall.EgressFirewall)
+			oldEgressFirewall := old.(*egressfirewall.EgressFirewall)
+			if !reflect.DeepEqual(oldEgressFirewall, newEgressFirewall) {
```

Comment:
What happens here when you update the status on the egress firewall, smells like an infinite loop? Shouldn't this check be:

```
!reflect.DeepEqual(oldEgressFirewall.Spec, newEgressFirewall.Spec) 
```

Reply:
it would be, before I moved it out here it was checking the spec